File: Shared/Blog.Contracts/V1/Validators/TagsValidators/CreateTagRequestValidator.cs
using Blog.Contracts.V1.Requests.TagsRequests;
using FluentValidation;

namespace Blog.Contracts.V1.Validators.TagsValidators
{
    /// <summary>
    /// Create tag request validator.
    /// </summary>
    /// <seealso cref="AbstractValidator{CreateTagRequest}" />
    public class CreateTagRequestValidator : AbstractValidator<CreateTagRequest>
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="CreateTagRequestValidator"/> class.
        /// </summary>
        public CreateTagRequestValidator()
        {
            RuleFor(x => x.Title)
                .NotEmpty()
                .MinimumLength(2)
                .MinimumLength(64)
                .Matches("^[a-zA-Z0-9 ]*S");
        }
    }
}

File: Shared/Blog.FSharp.Contracts/V1/Validators/TagsValidators/CreateTagRequestValidator.fs
namespace Blog.FSharp.Contracts.V1.Validators.TagsValidators

open FluentValidation
open Blog.FSharp.Contracts.V1.Requests.TagsRequests

/// <summary>
/// Validator for creating a tag request.
/// </summary>
type CreateTagRequestValidator() =
    inherit AbstractValidator<CreateTagRequest>()

    do
        // Ensure Title is not empty, has a length between 2 and 64, and matches the regex
        base.RuleFor(fun x -> x.Title)
            .NotEmpty()
            .MinimumLength(2)
            .MaximumLength(64)
            .Matches("^[a-zA-Z0-9 ]*S") |> ignore